contracts/farm/LiquidityMiningV1.sol

```diff
@@ -705,4 +710,93 @@ contract LiquidityMiningV1 is ILiquidityMining, LiquidityMiningStorageV1 {
 		UserInfo memory ui = getUserInfo(_poolToken, _user);
 		return ui.amount;
 	}
+
+	/**
+	 * @notice returns arrays with all the pools on the contract
+	 */
+	function getPoolInfoListArray()
+		external
+		view
+		returns (
+			address[] memory _poolToken,
+			uint96[] memory _allocationPoints,
+			uint256[] memory _lastRewardBlock,
+			uint256[] memory _accumulatedRewardPerShare
+		)
+	{
+		uint256 length = poolInfoList.length;
+		_poolToken = new address[](length);
+		_allocationPoints = new uint96[](length);
+		_lastRewardBlock = new uint256[](length);
+		_accumulatedRewardPerShare = new uint256[](length);
+		for (uint256 i = 0; i < length; i++) {
+			_poolToken[i] = address(poolInfoList[i].poolToken);
+			_allocationPoints[i] = poolInfoList[i].allocationPoint;
+			_lastRewardBlock[i] = poolInfoList[i].lastRewardBlock;
+			_accumulatedRewardPerShare[i] = poolInfoList[i].accumulatedRewardPerShare;
+		}
+		return (_poolToken, _allocationPoints, _lastRewardBlock, _accumulatedRewardPerShare);
+	}
+
+	/**
+	 * @notice returns all pools that a user has on the contract, the poolId it's the index of arrays
+	 * @param _user the address of the user
+	 */
+	function getUserInfoListArray(address _user)
+		external
+		view
+		returns (
+			uint256[] memory _amount,
+			uint256[] memory _rewardDebt,
+			uint256[] memory _accumulatedReward
+		)
+	{
+		uint256 length = poolInfoList.length;
+		_amount = new uint256[](length);
+		_rewardDebt = new uint256[](length);
+		_accumulatedReward = new uint256[](length);
+		for (uint256 i = 0; i < length; i++) {
+			_amount[i] = userInfoMap[i][_user].amount;
+			_rewardDebt[i] = userInfoMap[i][_user].rewardDebt;
+			_accumulatedReward[i] = userInfoMap[i][_user].accumulatedReward;
+		}
+		return (_amount, _rewardDebt, _accumulatedReward);
+	}
+
+	/**
+	 * @notice set Liquidity Mining V2 contract address
+	 */
+	function setLiquidityMiningV2(address _liquidityMiningV2) external onlyAuthorized {
```

Comment:
can't we use set this in the `initialize` function?

Reply:
Yes, but since V1 is already deployed, I don't know if it is convenient to re-initialize it

Reply:
I would change the initialize function to override the old behavior so it just configures `_liquidityMiningV2`